src/app/AttributeAccessInterface.h

```diff
@@ -48,6 +50,36 @@ class AttributeAccessInterface
     {}
     virtual ~AttributeAccessInterface() {}
 
+    /**
+     * Allows a provider to register itself as using this attribute access interface.
+     *
+     * The intent is to be able to hook into:
+     *   - interactions with other clusters (ability to read/write/invoke)
+     *   - Access to `CurrentContext()` to generate event and (RARELY) use an ExchangeContext
+     *      - NOTE: usage of dataModelChangeListerner would only trigger path dirty sets
+     *              however the DAtaModel::Provider manages cluster versions. As a result,
+     *              the `ChangeListener` passed in should be used instead to both modify
+     *              cluster version and trigger path dirtying
+     *   - nofify of data model changes (which will update data model dirty paths)
+     *
+     * @param [in] provider - use it for read/write/invoke and for GetCurrentContext
+     * @param [in] attributeChangeListener - use this to `MarkDirty` a specific parth that
+     *        this AttributeAccessInterfaces manages. The paths SHOULD match mEndpointID/mClusterId
+     *        for this AttributeAccessInterface
+     *
+     * Lifetime of provider and attributeChangeListener is valid until DetachProvider is called
+     * with the same provider value.
+     */
+    virtual void AttachToProvider(DataModel::Provider * provider, DataModel::ProviderChangeListener * attributeChangeListener) {}
```

Comment:
I don't think this really scales once we have multiple providers around, including some that delegate some endpoints to one provider and some to another provider, possibly with the same cluster on both sets of endpoints using a single AttributeAccessInterface.

Put another way, this breaks as soon as you have a bridge that wants to use ZAP for its static features and a different provider for the dynamic bridge functionality.

I suspect the right way to do this is to pass the relevant context to the AttributeAccessInterface via whatever APIs are going to need to have that information eventually, not this binding setup.

Reply:
Regarding multiple providers, I believe long term the providers should handle their attribute access interfaces or equivalent. Having a single `AttributeAccessInterfaceRegistry` and similar for commands is essentially a global singleton and that starts forcing singletons within our providers and that is too much coupling.

Since the InteractionModelEngine logic now does not directly call AAI/CHI anymore, I believe the providers should handle their AAI. If an application has multiple providers, those providers should manage their AAI.